lib/finch/http1/conn.ex

```diff
@@ -174,33 +175,83 @@ defmodule Finch.Conn do
         Telemetry.stop(:recv, start_time, metadata, extra_measurements)
         {:ok, %{conn | mint: mint}, acc}
 
+      {:error, _mint, error = :response_body_too_large} ->
+        metadata = Map.put(metadata, :error, error)
+        Telemetry.stop(:recv, start_time, metadata, extra_measurements)
+        {:error, close(conn), Error.exception(error)}
+
       {:error, mint, error} ->
         metadata = Map.put(metadata, :error, error)
         Telemetry.stop(:recv, start_time, metadata, extra_measurements)
         {:error, %{conn | mint: mint}, error}
```

Comment:
Should you close the connection here? I closed it above because the subsequent request would fail if I didn't.